components/camel-jaxb/src/main/java/org/apache/camel/converter/jaxb/JaxbDataFormat.java

```diff
@@ -258,6 +238,27 @@ void doMarshal(Exchange exchange, Object graph, OutputStream stream, Marshaller
         }
     }
 
+    private void performWrite(Exchange exchange, OutputStream stream, XMLStreamWriter writer, Marshaller marshaller, Object toMarshall) throws JAXBException {
+        if (asXmlStreamWriter(exchange)) {
+            marshaller.marshal(toMarshall, writer);
+        } else {
+            marshaller.marshal(toMarshall, stream);
+        }
+    }
+
```

Comment:
With the proposed refactoring, getWriter is called at each marshall attempt (performance) and even when `asXmlStreamWriter(exchange)` is false (behavior change). I would not include this change.

Reply:
Thanks, missed that. I'll revert it.